keel-core/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/NetflixSemVerVersioningStrategy.kt

```diff
@@ -0,0 +1,47 @@
+package com.netflix.spinnaker.keel.artifacts
+
+import com.netflix.spinnaker.keel.api.artifacts.PublishedArtifact
+import com.netflix.spinnaker.keel.api.artifacts.VersioningStrategy
+import com.netflix.spinnaker.keel.core.NETFLIX_SEMVER_COMPARATOR
+
+/**
+ * A [VersioningStrategy] that codifies the versioning scheme conventions at Netflix.
+ */
+object NetflixSemVerVersioningStrategy : VersioningStrategy {
+  override val type: String = "netflix-semver"
```

Comment:
Technically, this is used as the discriminator to determine the sub-type of `VersioningStrategy` for deserialization, but we don't expose that in the delivery config as a property (it's hard-coded with each `DeliveryArtifact` sub-class), so this type name shouldn't matter.